src/qcodes/dataset/database_extract_runs.py

```diff
@@ -168,3 +175,229 @@ def _extract_single_dataset_into_db(
     _populate_results_table(
         source_conn, target_conn, dataset.table_name, target_table_name
     )
+
+
+def export_datasets_and_create_metadata_db(
+    source_db_path: str | Path,
+    target_db_path: str | Path,
+    export_path: str | Path | None = None,
+    upgrade_source_db: bool = False,
+    upgrade_target_db: bool = False,
+) -> dict[int, str]:
+    """
+    Export all datasets from a source database to NetCDF files and create
+    a new database file containing only metadata (no raw data) for those exported
+    datasets. Datasets that cannot be exported to NetCDF will be transferred
+    as-is to the new database.
+
+    This function is useful for reducing the size of database files by offloading
+    raw data to NetCDF files while preserving all metadata and structural information.
+
+    Args:
+        source_db_path: Path to the source database file
+        target_db_path: Path to the target database file. Will be created if it doesn't exist.
+        export_path: Optional path where NetCDF files should be exported. If None,
+            uses the default export path from QCoDeS configuration.
+        upgrade_source_db: If the source DB is found to be in a version that is
+            not the newest, should it be upgraded?
+        upgrade_target_db: If the target DB is found to be in a version that is
+            not the newest, should it be upgraded?
+
+    Returns:
+        A dictionary mapping run_id to status ('exported' or 'copied_as_is')
+
+    Raises:
+        ValueError: If there are issues with the database files or datasets
+    """
+    # Convert paths to Path objects
+    source_db_path = Path(source_db_path)
+    target_db_path = Path(target_db_path)
+    
+    if export_path is None:
+        export_path = get_data_export_path()
+    else:
+        export_path = Path(export_path)
+    
+    # Validate source database exists
+    if not source_db_path.exists():
+        raise FileNotFoundError(f"Source database file not found: {source_db_path}")
+    
+    log.info(f"Starting export process from {source_db_path} to {target_db_path}")
+    log.info(f"NetCDF files will be exported to {export_path}")
+    
+    # Check database versions
+    try:
+        (s_v, new_v) = get_db_version_and_newest_available_version(source_db_path)
+        if s_v < new_v and not upgrade_source_db:
+            warn(
+                f"Source DB version is {s_v}, but this function needs it to be"
+                f" in version {new_v}. Run this function again with "
+                "upgrade_source_db=True to auto-upgrade the source DB file."
+            )
+            return {}
+    except Exception as e:
+        log.error(f"Failed to check source database version: {e}")
```

Comment:
use log.exception, it's much neater

Reply:
go also through all places in the new code where exception is being logged and make the corresponding change